main.cpp

```diff
@@ -369,116 +372,119 @@ struct LogicBoard
     std::string nameOfDeveloper = "Nintendo";
-//     5) Price of game (float)
     float priceOfGame = 19.99f;
-//     1) Load levels
     void loadLevels(int levelNumber, float gameStatus);
-//     2) Take input from controls
     void takeControlInput(Controls currentInput);
-//     3) Output video signals
-    float outputVideoSignal(float gameStatus, int numberOfPixelsUsed, Display targetDisplay); // outputs magical float signal that the logic board could use.
+    float outputVideoSignal(float gameStatus, int numberOfPixelsUsed, Display targetDisplay);
 };
 
-/*
-Thing 9) Outside Case
-    1) Color (std::string)
-    2) Width (float)
-    3) Height (float)
-    4) Depth (float)
-    5) Number of LED lights (int)
-    1) Protect insides
-    2) Display cool color
-    3) Light up
- */
+void Game::loadLevels(int levelNumber, float gameStatus)
+{
+    std::cout << "Loading level " << levelNumber << ". Game status is " << gameStatus << std::endl;
+}
+
+void Game::takeControlInput(Controls currentInput)
+{
+    std::cout << "Currently using " << currentInput.numberOfOtherButtons << " buttons." << std::endl;
+}
+
+float Game::outputVideoSignal(float gameStatus, int numberOfPixelsUsed, Display targetDisplay)
+{
+    return (gameStatus * targetDisplay.numberOfPixels) - numberOfPixelsUsed;
+}
 
+// --------------------------------------------------------------------------
 struct OutsideCase
 {
-//     1) Color (std::string)
     std::string color = "grey";
-//     2) Width (float)
     float width = 6.0f;
-//     3) Height (float)
     float height  = 12.2f;
-//     4) Depth (float)
     float depth = 1.0f;
-//     5) Number of LED lights (int)
-    int numberOfLEDs = 2;
-//     1) Protect insides
+    int totalNumberOfLEDs = 2;
     void protectInsides(int numberOfComponents, LogicBoard theLogicBoard, Display currentDisplay, Controls theControls);
-//     2) Display cool color
     void displayCoolColor(std::string newColor, int numberOfLEDs);
-//     3) Light up
     void lightUpLEDs(int numberOfLEDs);
 };
 
-/*
-Thing 10) Gameboy
-    1) Display
-    2) Controls
-    3) Logic Board
-    4) Game
-    5) Outside Case
-    1) Show visuals
-    2) Play sounds
-    3) Get input from user
- */
+void OutsideCase::protectInsides(int numberOfComponents, LogicBoard theLogicBoard, Display currentDisplay, Controls theControls)
+{
+    std::cout<< "Currently Protecting " << numberOfComponents << " components. Also protecting " << theLogicBoard.speedOfCPU << " and " << currentDisplay.numberOfPixels << " and " << theControls.numberOfOtherButtons << std::endl; 
+}
+
+void OutsideCase::displayCoolColor(std::string newColor, int numberOfLEDs)
+{
+    std::cout << "Change " << numberOfLEDs << " LEDs to color: " << newColor << std::endl;
+}
 
+void OutsideCase::lightUpLEDs(int numberOfLEDs)
+{
+    for(int i = 0; i < numberOfLEDs; i++)
+        {
+            std::cout << "Shine on your crazy LED." << std::endl;
+        }
+}
```

Comment:
loops have not been covered yet.
Please remove this and write something else instead.
- [ ] revise